internal/manifests/collector/adapters/config_to_ports.go

```diff
@@ -58,3 +65,89 @@ func ConfigToMetricsPort(logger logr.Logger, config map[interface{}]interface{})
 
 	return int32(i64), nil
 }
+
+func GetServicePortsFromCWAgentOtelConfig(logger logr.Logger, config map[interface{}]interface{}) ([]corev1.ServicePort, error) {
+	ports, err := ConfigToComponentPorts(logger, ComponentTypeReceiver, config)
+	if err != nil {
+		logger.Error(err, "there was a problem while getting the ports from the receivers")
+		return nil, err
+	}
+
+	sort.Slice(ports, func(i, j int) bool {
+		return ports[i].Name < ports[j].Name
+	})
+
+	return ports, nil
+}
+
+// ConfigToComponentPorts converts the incoming configuration object into a set of service ports required by the exporters.
+func ConfigToComponentPorts(logger logr.Logger, cType ComponentType, config map[interface{}]interface{}) ([]corev1.ServicePort, error) {
+	// now, we gather which ports we might need to open
+	// for that, we get all the exporters and check their `endpoint` properties,
```

Comment:
Why do we need to open ports for exporters? Id expect we only listen on ports for receivers.

Reply:
Ports only opened for receivers. Updated the comment to reflect the same

Reply:
nit: The comments for and in this function still refer to exporters.